mmuser.go

```diff
@@ -22,8 +22,11 @@ func NewUserMM(c net.Conn, srv Server, cfg *MmCfg) *User {
 	u.Srv = srv
 	u.MmInfo.Cfg = cfg
 
+	if cfg.LoginServiceNick == nil {
```

Comment:
Did you actually test it ? :-) 
Should be changed to == "" instead of == nil (LoginServiceNick is a string)